src/inmanta/agent/handler.py

```diff
@@ -179,6 +179,30 @@ def __init__(
         else:
             self.logger = logger
 
+        self._facts: List[Dict[str, Any]] = []
+
+    def set_fact(self, fact_id: str, value: str, metadata: dict = {}) -> None:
+        """
+            Send a fact to the Inmanta server.
+
+            :param fact_id: The name of the fact.
+            :param value: The actual value of the fact.
+            :param metadata: Metadata about the fact.
+        """
+        resource_id = self._resource.id.resource_str()
+        fact = {
+            "id": fact_id,
+            "source": "agent",
+            "value": value,
+            "resource_id": resource_id,
+            "metadata": metadata,
```

Comment:
@bartv is the metadata ever used? I can't immediately find a place where is is used.

Reply:
Generic parameters sometimes have used facts. For example the first version of the lifecycle mgmt. However, there are no facts that ever used this.

Reply:
So I remove the metadata parameter from the `set_fact()` method in the HandlerContext?

Reply:
yes, I do not see a use case for it at the moment. Adding parameters later is easy, removing them is a lot harder :)